http/examples/http_meta.rs

```diff
@@ -1,8 +1,10 @@
 extern crate jsonrpc_http_server;
+extern crate unicase;
 
-use jsonrpc_http_server::{ServerBuilder, hyper, RestApi};
+use jsonrpc_http_server::{ServerBuilder, hyper, RestApi, AccessControlAllowHeaders};
 use jsonrpc_http_server::jsonrpc_core::*;
 use self::hyper::header;
+use self::unicase::Ascii;
```

Comment:
IMHO we should use re-exported value - actually user doesn't really care that it's `Ascii` internally, can't we do like this:
```rust
AccessControlAllowHeaders::Only(vec![
  "Authorization",
])
```
and convert inside the builder?

Reply:
Ah yeah, I agree, the API should only expose what is necessary.

I added a commit which introduces a struct `AccessControlAllowHeadersUnicase` which is a `HashSet<Ascii<&'static str>>` and is used internally now. I added an `Into` trait which converts the exposed `AccessControlAllowHeaders` to this internal type, so it's still possible to use a Vector for setting everything up.